sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -695,7 +695,12 @@ fn type_check_size_of_type(
             type_id,
             initial_type_id,
             span: targ.span(),
-            call_path_tree: targ.call_path_tree().cloned(),
+            call_path_tree: targ
+                .as_type_argument()
+                .unwrap()
+                .call_path_tree
+                .as_ref()
+                .cloned(),
```

Comment:
This pattern is repeating a lot of times, in this and other files so it somehow calls for abstraction. Can this be encapsulated behind a method on `GenericArgument`? Like `GenericArgument::type_arg_call_path_tree` or similar?

Reply:
Let me analyse this in another PR. The unwrap was on purpose because these snippets only work for type generics.... and not for const generics.

Reply:
I think the issue may be caused by using `GenericArgument` when it is not needed. Example `TypeInfo::Ptr(GenericArgument)`. The older method was hiding this. 

This was one of the reasons for removing them.

I will try to fix it in future PRs. Using the correct `GenericTypeArgument` should eliminate this horrible "pattern".